Source/Graphic/HCPath.c

```diff
@@ -349,87 +658,161 @@ void HCPathAddElement(HCPathRef self, HCPathCommand command, const HCPoint* poin
+    // Determine if there are elements to remove
     HCInteger elementCount = HCPathElementCount(self);
+    if (elementCount == 0) {
+        return;
+    }
     
-    // Remove element
+    // Find the element to be removed
     HCPathElement element = HCPathElementAt(self, elementCount - 1);
-    if (element.points != NULL) {
-        free(element.points);
+    
+    // Remove corresponding property from contour
+    switch (element.command) {
+        case HCPathCommandMove: {
+            HCListRemoveObject(self->contours);
+        } break;
+        case HCPathCommandAddLine:
+        case HCPathCommandAddQuadraticCurve:
+        case HCPathCommandAddCubicCurve:
+        case HCPathCommandCloseContour: {
+            HCDataRef contourData = HCPathCurrentContour(self);
+            if (element.command != HCPathCommandCloseContour) {
+                HCDataRemoveBytes(contourData, sizeof(HCContourCurve));
+            }
+            HCInteger curveCount = HCDataSize(contourData) / sizeof(HCContourCurve);
+            if (curveCount == 0) {
+                HCListRemoveObject(self->contours);
+            }
+            else {
+                HCContourInitInCurves((HCContourCurve*)HCDataBytes(contourData), curveCount, false);
+            }
+        } break;
     }
-    HCDataRemoveBytes(self->elementData, sizeof(HCPathElement));
     
     // Remove element polyline data
-    HCListRemoveObject(self->elementPolylines);
+    HCListRemoveObject(self->polylines);
     
-    // Update count
+    // Remove element
+    if (element.points != NULL) {
+        free(element.points);
+    }
+    HCDataRemoveBytes(self->elementData, sizeof(HCPathElement));
     elementCount = HCPathElementCount(self);
     
     // Re-calculate bounds to exclude the element polylines
     HCRectangle bounds = HCRectangleMake(HCPathCurrentPoint(self), HCSizeZero);
     for (HCInteger elementIndex = 0; elementIndex < elementCount; elementIndex++) {
-        HCInteger polylinePointCount = HCPathElementPolylinePointCount(self, elementIndex);
+        HCInteger polylinePointCount = HCPathPolylinePointCount(self, elementIndex);
         for (HCInteger pointIndex = 0; pointIndex < polylinePointCount; pointIndex++) {
-            HCPoint point = HCPathElementPolylinePointAt(self, elementIndex, pointIndex);
+            HCPoint point = HCPathPolylinePointAt(self, elementIndex, pointIndex);
             bounds = HCRectangleIncludingPoint(bounds, point);
         }
     }
     self->bounds = bounds;
 }
 
 //----------------------------------------------------------------------------------------------------------------------------------
-// MARK: - Polylines
+// MARK: - Polyline Conversion
 //----------------------------------------------------------------------------------------------------------------------------------
 void HCPathAddLineSegmentPolylineData(HCPathRef self, HCReal x0, HCReal y0, HCReal x1, HCReal y1, HCDataRef polylineData) {
+    (void)self; // unused
```

Comment:
Is this needed?

Reply:
Yes, otherwise compiling with -Wall gives a warning.